### PR TITLE
fix(api-extension): Handle correct array value

### PR DESCRIPTION
**Pull request by jiqiang90**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1130 

## What is the new behavior?

Correctly handle array in ApiExtension

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,17414 +1,8 @@
 {
   "name": "@nestjs/swagger",
   "version": "4.7.12",
-  "lockfileVersion": 2,
```
Comment (coyotte508) [inline package-lock.json:4]: Do not edit this file - no modules are added or removed or updated

File: lib/decorators/api-extension.decorator.ts [hunk 2]
```diff
@@ -9,10 +10,7 @@ export function ApiExtension(extensionKey: string, extensionProperties: any) {
   }
 
   const extensionObject = {
-    [extensionKey]:
-      typeof extensionProperties !== 'string'
-        ? { ...extensionProperties }
-        : extensionProperties
+    [extensionKey]: clone(extensionProperties)
```
Comment (coyotte508) [inline lib/decorators/api-extension.decorator.ts:13]: The tests should probably be changed (e.g. the e2e tests) to add an example with an array value

**Review by coyotte508 — CHANGES_REQUESTED** [review]

File: lib/decorators/api-extension.decorator.ts (hunk 2 above)
Comment (jiqiang90) [reply lib/decorators/api-extension.decorator.ts:13]: Test case added

**Review by coyotte508 — APPROVED** [review]

**Comment by ianhe8x** [comment]
This feature is blocking us, may I ask when it will get merged?

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**